cmd/lakectl/cmd/fs.go

```diff
@@ -115,28 +115,31 @@ var fsCatCmd = &cobra.Command{
 	Short: "Dump content of object to stdout",
 	Args:  cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		client := getClient()
 		pathURI := MustParsePathURI("path", args[0])
 		direct, _ := cmd.Flags().GetBool("direct")
-		var contents []byte
+		var err error
+		var body io.ReadCloser
 		if direct {
-			_, body, err := helpers.ClientDownload(cmd.Context(), client, pathURI.Repository, pathURI.Ref, *pathURI.Path)
+			client := getClient()
+			_, body, err = helpers.ClientDownload(cmd.Context(), client, pathURI.Repository, pathURI.Ref, *pathURI.Path)
 			if err != nil {
 				DieErr(err)
 			}
 			defer body.Close()
-			contents, err = io.ReadAll(body)
-			if err != nil {
-				DieErr(err)
-			}
 		} else {
-			resp, err := client.GetObjectWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.GetObjectParams{
+			client := getClientWithoutResponse()
```

Comment:
I suspect that the client fully downloads the file here (otherwise we'd probably have Close'd the Body). Best look at the generated code - note that this path is the common one for our users).

Reply:
It does not download, it returns an HTTP response with a reader object. The file should indeed be closed. Fixed.